tests/main/basic-target-socket-activation/task.yaml

```diff
@@ -0,0 +1,74 @@
+summary: Check that mounts are mounted correctly if snapd gets activated
+
+details: |
+  When booting with basic.target for example, snap mounts are not
+  mounted.  But sockets.target is active, which means snapd.socket
+  is. If a snap command is emitted, then snapd will start. But it
+  expects mounts to be there.
+
+systems: [ubuntu-2*]
+
+prepare: |
+  cat <<\EOF >/etc/default/grub.d/99-basic.cfg
+  # Use basic.target as default target
+  GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} systemd.unit=basic.target"
+  # spread will need to reconnnect to ssh
+  GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} systemd.wants=ssh.service"
+  # ... through network
+  GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} systemd.wants=systemd-networkd.service"
+  # .. as a non-system user
+  GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} systemd.wants=systemd-user-sessions.service"
+  EOF
+
+  update-grub
+
+restore: |
+  if [ "${SPREAD_REBOOT}" = 0 ]; then
+
+    rm -f /etc/default/grub.d/99-basic.cfg
+    update-grub
+
+    if grep 'systemd[.]unit=basic[.]target' /proc/cmdline; then
+      REBOOT
+    fi
+
+  fi
+
+execute: |
+  if [ "${SPREAD_REBOOT}" = 0 ]; then
+
+    REBOOT
+
+  else
+
+    # Wait for systemd to have finished booting
+    systemctl --wait is-system-running | MATCH 'running'
+
+    MATCH 'systemd[.]unit=basic[.]target' </proc/cmdline
+
+    # basic.target and snapd.socket should be active
+    systemctl show -p ActiveState basic.target | MATCH 'ActiveState=active'
+    systemctl show -p ActiveState snapd.socket | MATCH 'ActiveState=active'
+
+    # ... but snapd.service and  multi-user.target should not.
+    systemctl show -p ActiveState multi-user.target | MATCH 'ActiveState=inactive'
+    systemctl show -p ActiveState snapd.service | MATCH 'ActiveState=inactive'
+
+    # Let's emit a snap command. That will start snapd.
+    snap list
+
+    # multi-user should still be inactive but snapd active.
+    systemctl show -p ActiveState multi-user.target | MATCH 'ActiveState=inactive'
+    systemctl show -p ActiveState snapd.service | MATCH 'ActiveState=active'
+
+    # the mounts should be active
+    systemctl show --all -p ActiveState 'snap-*.mount' | tee states.txt | MATCH 'ActiveState=active'
+    NOMATCH 'ActiveState=inactive' <states.txt
+
+    # bonus, we check we did not cheat, nothing should be installed to default.target
+    # shellcheck disable=SC2251
+    ! [ -d /etc/systemd/system/default.target.wants ]
```

Comment:
nitpick: you could use test -d <DIR> instead of [ -d <DIR> ]

Reply:
also in tests you can use not instead of !

Reply:
Interesting. I will use `not`.